xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DocumentsMacro.xml

```diff
@@ -256,65 +259,53 @@ Generates:
-  'rowCount' : $xcontext.macro.params.count,
-  'description' : 'This table lists documents found on this wiki based on passed criteria. The columns can be sorted and some can be filtered.'
-})
-#if ("$!xcontext.macro.params.space" != "") 
-  #set ($extraParams = "&amp;space=$escapetool.url($xcontext.macro.params.space)")
-#end
-#if ("$!xcontext.macro.params.location" != "") 
-  #set ($extraParams = "$!{extraParams}&amp;location=$escapetool.url($xcontext.macro.params.location)")
-#end
-#if ("$!xcontext.macro.params.parent" != "") 
-  #set ($extraParams = "$!{extraParams}&amp;parent=$escapetool.url($xcontext.macro.params.parent)")
+#if ("$!xcontext.macro.params.space" != "")
+  #set ($discard = $sourceParameters.put('space', $xcontext.macro.params.space))
 #end
-#if ("$!extraParams" != "")
-  #set($discard = $options.put('extraParams', $extraParams))
+#if ("$!xcontext.macro.params.location" != "")
+  #set ($discard = $sourceParameters.put('location', $xcontext.macro.params.location))
 #end
-#if(!$isGuest &amp;&amp; $xcontext.macro.params.actions == "true")
-  #set($discard = $collist.add('_actions'))
-  #set($discard = $colprops.put('_actions', {
-    'actions': ['copy', 'rename', 'rights', 'delete'],
-    'filterable': false,
-    'sortable': false
-  }))
+#if ("$!xcontext.macro.params.parent" != "")
+  #set ($discard = $sourceParameters.put('parent', $xcontext.macro.params.parent))
 #end
+#set ($propertyDescriptors = [])
 #if ("$!services.like" != "")
-  #set($discard = $collist.add('_likes'))
   ## We can't make it sortable or filterable right now since it's not stored in DB.
-  #set($discard = $colprops.put('_likes', {
-    'filterable': false,
-    'sortable': false
-  }))
+  #set ($discard = $propertyDescriptors.add({ 'id': '_likes', 'sortable': false, 'filterable': false }))
+  #set ($discard = $properties.add('_likes'))
 #end
-#if ($collist.contains('doc.location'))
-  #set ($discard = $options.put('selectedColumn','doc.location'))
+#if(!$isGuest &amp;&amp; $xcontext.macro.params.actions == "true")
+  #set ($discard = $propertyDescriptors.add({ 'id': '_actions', 'displayer': { 'id': 'actions', 'actions': ['copy', 'rename', 'rights', 'delete'] }}))
+  #set ($discard = $properties.add('_actions'))
 #end
+#set ($liveDataConfig = {'meta': {'propertyDescriptors': $propertyDescriptors}})
 #if ("$!xcontext.macro.params.id" != "")
-  #set ($livetableId = $xcontext.macro.params.id)
+  #set ($livedataId = $xcontext.macro.params.id)
 #else
   ## TODO: Improve this since we could have several livetables on the same page with same ids
-  #set ($livetableId = "documents-$mathtool.random(1, 1000)")
+  #set ($livedataId = "documents-$mathtool.random(1, 1000)")
 #end
-#livetable($livetableId $collist $colprops $options)
+
+{{liveData
+  id="$livedataId"
+  properties="$stringtool.join($properties, ',')"
+  source="liveTable"
+  sourceParameters="$escapetool.url($sourceParameters)"
+  description="$!xcontext.macro.params.description"
```

Comment:
If the parameter is empty we don't get anything in the dom for the description right? i.e. it has 0 impact for backward compat' of the macro?

Reply:
The old behavior was having a useless and generic "This table lists documents found on this wiki based on passed criteria. The columns can be sorted and some can be filtered." message in the dom (but hidden with a `sr-only` class) for all Livetable generated through the `documents` macro.
Now, a description is inserted only if passed explicitly to the `description` parameter of the `documents` macro.
This can be considered as a breaking change as the default message is not there anymore, but imo this is also an accessibility improvement as we remove some useless noise for users using screen readers.